### DISCUSSION TITLE
false alert by rb/incomplete-hostname-regexp

### DISCUSSION THREAD

**Discussion by nobu** [github/codeql#9887] [Q&A]
Recently, alerts like https://github.com/ruby/ruby/security/code-scanning/179 are detected.
However, these suspected strings are not used as regexps, but are argument strings for `URI::RFC3986_Parser::RFC3986_URI.match` and `URI::RFC3986_Parser::RFC3986_relative_ref.match`, which are `Regexp#match`.

**Comment by edoardopirovano** [comment]
Many thanks for reporting this! Our Ruby support is still in beta, so we're still ironing out some issues in it. I'll pass this along to the relevant team to take a look at.
